### PR TITLE
Branch 2

### LINKED ISSUE #2: Finalize tasks for branch_2
- commit all changes in branch_2
- push changes to origin
- create pull request branch_2 > master

### PR DESCRIPTION
**Pull request by DrVladislove**
fixes #2

### CODE REVIEW

File: test/operations-conditions-1.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+var assert = require("assert");
```
Comment (spetrunin) [inline test/operations-conditions-1.js:1]: no need to have require in files itself, it will be enough to add it in test file

File: test/operations-conditions-1.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+function greaterNum(num1, num2) {
+  if (num1 > num2) {
```
Comment (spetrunin) [inline test/operations-conditions-1.js:3]: this condition should be simplified if u would take into account equal condition

File: test/operations-conditions-2.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+var assert = require("assert");
```
Comment (spetrunin) [inline test/operations-conditions-2.js:1]: same here

File: test/operations-conditions-3.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+var assert = require("assert");
```
Comment (spetrunin) [inline test/operations-conditions-3.js:1]: same here

File: test/operations-loops-1.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+var assert = require("assert");
+function randomizer(min, max) {
+  return Math.floor(min + Math.random() * (max + 1 - min));
+}
+function numberGenerator() {
+  for (var i = 0; i < 5; i++) {
+    var a = randomizer(1, 20);
+    if (a % 2 == 0) {
+      return a, "Четное";
+    } else if (a % 2) {
```
Comment (spetrunin) [inline test/operations-loops-1.js:10]: it could be just simple
```
} else {
```

File: test/operations-loops-3.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+var assert = require("assert");
```
Comment (spetrunin) [inline test/operations-loops-3.js:1]: completely wrong

here you should require your existing solution
```
 test/operations-conditions-3.js
```

and call function from it in your loop

File: test/test-conditions-1.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
+const operations = require("./operations-conditions-1.js");
+const assert = require("assert");
+
+it("Правильно определяет условие conditions-task-1", () => {
```
Comment (spetrunin) [inline test/test-conditions-1.js:4]: For better test coverage you should have tests for edge cases like:
-`-1` and  `0`
- `0` and `1`
- `1` and `1`

File: test/test-conditions-2.js [hunk 8]
```diff
@@ -0,0 +1,6 @@
+const operations = require("./operations-conditions-2.js");
+const assert = require("assert");
+
+it("Выдает правильный язык conditions-task-2", () => {
```
Comment (spetrunin) [inline test/test-conditions-2.js:4]: you have tested only one possible input, you should test all of them in this case

File: test/test-conditions-3.js [hunk 9]
```diff
@@ -0,0 +1,5 @@
+const operations = require("./operations-conditions-3.js");
+const assert = require("assert");
+it("Правильно определяет оценки conditions-task-3", () => {
+  assert.equal(operations.assignGrade(62, 75), "mark is 4");
```
Comment (spetrunin) [inline test/test-conditions-3.js:4]: you need more test cases here

Comment (spetrunin) [reply test/test-conditions-3.js:4]: Why you are passing 2 arguments here?
```
operations.assignGrade(62, 75)
```

When your function expects only one and will ignore additional

File: test/test-loops-1.js [hunk 10]
```diff
@@ -0,0 +1,9 @@
+const operations = require("./operations-loops-1.js");
+const assert = require("assert");
+
+it("Правильно находит Нечетные loops-task-1", () => {
+  assert.equal(operations.numberGenerator(), "Нечетное");
```
Comment (spetrunin) [inline test/test-loops-1.js:5]: test will fail randomly

File: test/test-loops-1.js [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
+});
+it("Правильно находит Четные loops-task-1", () => {
```
Comment (spetrunin) [inline test/test-loops-1.js:7]: test will fail randomly

File: test/operations-loops-2.js [hunk 12]
```diff
@@ -0,0 +1,12 @@
+var assert = require("assert");
+function multiplicationTable() {
```
Comment (spetrunin) [inline test/operations-loops-2.js:2]: your solution will return only a single line, not the full table

File: test/test-loops-2.js [hunk 13]
```diff
@@ -0,0 +1,5 @@
+const operations = require("./operations-loops-2.js");
+const assert = require("assert");
+it("Таблица умножения выполнена правильно loops-task-2", () => {
+  assert.equal(operations.multiplicationTable(), "1*1=1");
```
Comment (spetrunin) [inline test/test-loops-2.js:4]: You should check my tests with sinon for this one checks

File: test/test-loops-1.js (hunk 11 above)
Comment (spetrunin) [reply test/test-loops-1.js:7]: You should check my tests with sinon for this one checks

File: test_2_option.js [hunk 14]
```diff
@@ -0,0 +1,118 @@
+const expect = require("expect");
+
+function greaterNum(num1, num2) {
```
Comment (spetrunin) [inline test_2_option.js:3]: you should not copy functions here, instead you should require them from the corresponding file

**Review by devsergiy — CHANGES_REQUESTED** [review]
you should care more about quality and ask more questions

File: test/operations-conditions-1.js (hunk 2 above)
Comment (DrVladislove) [reply test/operations-conditions-1.js:3]: Do you mean rewrite condition? I didn't understand till the end this correction

Comment (spetrunin) [reply test/operations-conditions-1.js:3]: You do not care now about equal

For example, you could make one of condition inclusion with `<=` and in this case, another branch of `IF` could be simplified to simple else instead of `else if`

File: test/operations-conditions-2.js [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+  switch (lang) {
+    case "en":
+      return "Hello";
+    case "es":
+      return "Hola";
+    default:
+      return "Привет";
+  }
+}
+console.log(helloWorld());
```
Comment (spetrunin) [inline test/operations-conditions-2.js:16]: in real projects - such log messaged considered as debugging info and it should not be present in the final version

File: test/operations-conditions-1_test.js [hunk 16]
```diff
@@ -0,0 +1,15 @@
+/*1. Какое из чисел больше?
```
Comment (spetrunin) [inline test/operations-conditions-1_test.js:1]: you could skip task definition for tests, please include it only in task files

File: test/operations-conditions-1_test.js [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+Напишите функцию greaterNum которая: принимает 2 аргумента(числа).
+Bозвращает наибольшее число из полученных в аргументы.
+ Вызовите два раза с разными парами чисел, и выведите в консоль (например, "Наибольшее число между 5 и 10: 10.").
+*/
+
+const operations = require("./operations-conditions-1.js");
+const assert = require("assert");
+
+it("Правильно определяет условие conditions-task-1", () => {
```
Comment (spetrunin) [inline test/operations-conditions-1_test.js:10]: Better to arrange test cases in the next way

```javascript
describe("greaterNum", ()=> {
  it("Правильно определяет условие conditions-task-1", () => {
    assert.equal(operations.greaterNum(0, 1), 1);
  });

  it("Правильно определяет условие conditions-task-1", () => {
    assert.equal(operations.greaterNum(1, 1), 1);
  });
});
```

File: test/operations-conditions-1_test.js [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+  assert.equal(operations.greaterNum(0, 1), 1);
+});
+it("Правильно определяет условие conditions-task-1", () => {
```
Comment (spetrunin) [inline test/operations-conditions-1_test.js:13]: For better readability please make sure that you have an empty line between test cases

File: test/operations-conditions-1_test.js (hunk 17 above)
Comment (spetrunin) [reply test/operations-conditions-1_test.js:10]: Test case name should be more meaningful for example
```
it("returns the first argument when it is bigger than a second one"
```

File: test/operations-loops-1.js [hunk 19]
```diff
@@ -0,0 +1,22 @@
+/*1. Генератор четных - нечетных чисел
+Создайте цикл, который будет случайным образом генерировать 5 чисел от 0 до 20.
+ Для каждого случая будет проверятся текущее число (четное или нечетное), и выводится результат на экран (например, 2 - четное).
+ Решение оформите в виде функции numberGenerator()
+ */
+function randomizer(min, max) {
+  return Math.floor(min + Math.random() * (max + 1 - min));
+}
+function numberGenerator() {
```
Comment (spetrunin) [inline test/operations-loops-1.js:9]: Please add an empty line between functions

File: test/operations-loops-1.js [hunk 20]
```diff
@@ -0,0 +1,22 @@
...
+  for (var i = 0; i < 5; i++) {
+    var a = randomizer(1, 20);
+    if (a % 2 == 0) {
+      console.log(a + " четное");
+    } else {
+      console.log(a + " нечетное");
+    }
+  }
+}
+numberGenerator();
```
Comment (spetrunin) [inline test/operations-loops-1.js:19]: please remove debug call

File: test/operations-loops-2.js [hunk 21]
```diff
@@ -0,0 +1,17 @@
...
+
+function multiplicationTable() {
+  var resultString;
+  for (var i = 1; i <= 10; i++) {
+    for (var j = 1; j <= 10; j++) {
+      resultString = i + " * " + j + " = " + i * j;
+    }
+  }
+}
+multiplicationTable();
```
Comment (spetrunin) [inline test/operations-loops-2.js:14]: remove debug call

File: test/operations-loops-3.js [hunk 22]
```diff
@@ -0,0 +1,30 @@
+/* 3. Оценка результатов
+Cоздайте функцию checkAssignGrade:
+Проверьте результаты функции assignGrade из условных упражнений для каждого значения от 80 до 100.
+То есть создайте цикл for от 80 до 100 и вызывайте в нем вашу функцию, например assignGrade(81)
+*/
+
+function assignGrade(mark) {
```
Comment (spetrunin) [inline test/operations-loops-3.js:7]: instead of redefining function here - please require it

File: test/operations-loops-3.js [hunk 23]
```diff
@@ -0,0 +1,30 @@
...
+  }
+}
+
+function checkAssignGrade() {
+  for (i = 85; i <= 100; i++) {
+    console.log(assignGrade(81));
+    return i;
+  }
+}
+checkAssignGrade();
```
Comment (spetrunin) [inline test/operations-loops-3.js:27]: remove debug code

File: test/operations.js [hunk 24]
```diff
@@ -0,0 +1,10 @@
+var assert = require("assert");
+const multiply = (x, y) => x * y;
```
Comment (spetrunin) [inline test/operations.js:2]: please add spaces between `require` block and code itself

File: test/test-conditions-1.js (hunk 7 above)
Comment (spetrunin) [reply test/test-conditions-1.js:10]: edge cases are still not covered

File: test/test-conditions-2.js (hunk 8 above)
Comment (spetrunin) [reply test/test-conditions-2.js:8]: it's a duplicated test

File: test/test-conditions-3.js [hunk 25]
```diff
@@ -0,0 +1,20 @@
+/*3. Постановщик оценок
+Напишите функцию assignGrade которая: принимает один аргумент, значение успеваемости от 1 до 100. и возвращает число от 1 до 5: 5, 4, 3, 2, или 1 которое соответствует оценке
+*/
+const operations = require("./operations-conditions-3.js");
+const assert = require("assert");
+it("Правильно определяет оценки conditions-task-3", () => {
```
Comment (spetrunin) [inline test/test-conditions-3.js:6]: it's a duplicated test

File: test/test-loops-1.js (hunk 10 above)
Comment (spetrunin) [reply test/test-loops-1.js:10]: test duplicated or not correct

File: test/test-loops-2.js [hunk 26]
```diff
@@ -0,0 +1,45 @@
+function multiplicationTable() {
```
Comment (spetrunin) [inline test/test-loops-2.js:1]: seems duplicated file

make sure you do not have your functions redefined in test files
you should require them

File: test/test-loops-3.js [hunk 27]
```diff
@@ -0,0 +1,5 @@
+const operations = require("./operations-loops-3");
+const assert = require("assert");
+it("Оценка результатов выполнена правильно loops-task-3", () => {
+  assert.equal(operations.checkAssignGrade(), i);
```
Comment (spetrunin) [inline test/test-loops-3.js:4]: `i` variable is undefined
Also, it is not clear what you are testing here

File: test/test.js [hunk 28]
```diff
@@ -0,0 +1,14 @@
+const operations = require("./operations.js");
```
Comment (spetrunin) [inline test/test.js:1]: the filename is too generic

File: test/test_2_option.js [hunk 29]
```diff
@@ -0,0 +1,119 @@
+const expect = require("expect");
+var sinon = require("sinon");
+
+function greaterNum(num1, num2) {
```
Comment (spetrunin) [inline test/test_2_option.js:4]: same comment about functions redefined in test files

File: test_2_option.js (hunk 14 above)
Comment (spetrunin) [reply test_2_option.js:3]: not addressed

**Review by devsergiy — CHANGES_REQUESTED** [review]
Not all issues addressed, you should do more cleanup

And please make meaningful commit messages

File: test/test-conditions-2.js (hunk 8 above)
Comment (DrVladislove) [reply test/test-conditions-2.js]: What do you mean? Am I did test wrong or what?

File: test/operations-conditions-2_test.js [hunk 30]
```diff
@@ -0,0 +1,18 @@
...
+ По умолчанию возвращает русский вариант.
+ */
+const assert = require("assert");
+const operations = require("./operations-conditions-2.js");
+
+describe("Переводчик", function () {
+  it("Правильно определяет язык 'en' —> Hello", () => {
+    assert.equal(operations.helloWorld("en"), "Hello");
+  });
+  it("Правильно определяет язык 'es' —> Hola", () => {
```
Comment (spetrunin) [inline test/operations-conditions-2_test.js:12]: no blank lines between test cases

File: test/operations-loops-2.js [hunk 31]
```diff
@@ -0,0 +1,17 @@
+/*2. Таблица умножения
+Создайте цикл, который будет выполнятся от 1 до 10. Используйте вложенный цикл for, что бы показать таблицу для каждого множителя от 1 до 10 (всего 100 результатов). 
+Решение оформите в виде функции multiplicationTable
+*/
+
+function multiplicationTable() {
+  var resultString;
+  for (var i = 1; i <= 10; i++) {
+    for (var j = 1; j <= 10; j++) {
+      resultString = i + " * " + j + " = " + i * j;
```
Comment (spetrunin) [inline test/operations-loops-2.js:10]: you will override whole result string each time

Comment (spetrunin) [reply test/operations-loops-2.js:10]: and tests will not work for the current implementation

**Review by devsergiy — CHANGES_REQUESTED** [review]

File: loops-conditions/operations-loops-2.js [hunk 32]
```diff
@@ -0,0 +1,17 @@
+/*2. Таблица умножения
+Создайте цикл, который будет выполнятся от 1 до 10. Используйте вложенный цикл for, что бы показать таблицу для каждого множителя от 1 до 10 (всего 100 результатов). 
+Решение оформите в виде функции multiplicationTable
+*/
+
+function multiplicationTable() {
+  var resultString;
+  for (var i = 1; i <= 10; i++) {
+    for (var j = 1; j <= 10; j++) {
+      resultString = i + " * " + j + " = " + i * j;
```
Comment (spetrunin) [inline loops-conditions/operations-loops-2.js:10]: to fix test here should be next line
```
console.log(resultString);
```
